aiven/resource_service.go

```diff
@@ -839,6 +866,26 @@ func resourceServiceUpdate(ctx context.Context, d *schema.ResourceData, m interf
 		return diag.FromErr(err)
 	}
 
+	ass, dis, err := service.DiffStaticIps(ctx, d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var diags diag.Diagnostics
+	for i := range ass {
+		if err := client.StaticIPs.Associate(projectName, ass[i], aiven.AssociateStaticIPRequest{ServiceName: serviceName}); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
+	}
+	for i := range dis {
+		if err := client.StaticIPs.Dissociate(projectName, dis[i]); err != nil {
+			diags = append(diags, diag.FromErr(err)...)
+		}
+	}
+	if diags.HasError() {
+		return diags
+	}
+
```

Comment:
This logic of associating/dissociating the IPs is happening after the [UpdateServiceRequest](https://github.com/aiven/terraform-provider-aiven/blob/e04f29a135257ba1beb9f92ffdbe3524bb977deb/aiven/resource_service.go#L851-L864)   The static_ips=true flag attempts to be set the, but the association hasn't happened yet.

Reply:
fixing this locally seems to fix the update issue. I can see how this could get thru local testing if testing against a long-lived test service with static IPs already associated from another test (esp. since they cant be reliably dissociated currently :) ). Starting with a fresh service reproduced the issue reliably.

I moved the association of static IPs to happen before the service update, and dissociating static IPs to remain after the update (so that the static_ips can be updated to false prior to dissociating any IPs that were required while static_ips=true). Looks to be working good now.

Reply:
Hey really good catch, thanks for looking into it and testing!